packages/customize-widgets/src/components/sidebar-block-editor/sidebar-adapter.js

```diff
@@ -42,14 +68,20 @@ export default class SidebarAdapter {
 			),
 		];
 		this.historyIndex = 0;
+		this.historySubscribers = new Set();
+		// Debounce the input for 1 second.
+		this._debounceSetHistory = debounce(
+			this._pushHistory,
+			this._replaceHistory,
+			1000
+		);
 
 		this.setting.bind( this._handleSettingChange.bind( this ) );
 		this.api.bind( 'change', this._handleAllSettingsChange.bind( this ) );
 
-		this.canUndo = this.canUndo.bind( this );
-		this.canRedo = this.canRedo.bind( this );
```

Comment:
Don't we need to bind hasUndo/hasRedo too?

Reply:
Only if we're destructing it from the sidebar instance. Currently we're calling it directly so that we can keep referencing `this` to the instance.